qCC/ccCommandLineCommands.cpp

```diff
@@ -1324,70 +1324,50 @@ bool CommandSubsample::process(ccCommandLineInterface& cmd)
 			}
 
 			bool ok;
-			percent = cmd.arguments().takeFirst().toDouble(&ok);
+			const double percent = cmd.arguments().takeFirst().toDouble(&ok);
 			if (!ok || percent < 0 || percent > 100)
 			{
 				return cmd.error(QObject::tr("Invalid parameter: number after \"-%1 RANDOM %2\" must be decimal between 0 and 100").arg(COMMAND_SUBSAMPLE).arg(OPTION_PERCENT));
 			}
 
-			isPercent = true;
+			params = SubsamplingParams::RandomPercent(percent);
 		}
 		else
 		{
 			bool ok;
-			count = cmd.arguments().takeFirst().toUInt(&ok);
+			const unsigned count = cmd.arguments().takeFirst().toUInt(&ok);
 			if (!ok)
 			{
 				return cmd.error(QObject::tr("Invalid parameter: number of points or option \"%2\" after \"-%1 RANDOM \"").arg(COMMAND_SUBSAMPLE).arg(OPTION_PERCENT));
 			}
 			cmd.print(QObject::tr("\tOutput points: %1").arg(count));
+
+			params = SubsamplingParams::Random(count);
 		}
 		
 		for (CLCloudDesc& desc : cmd.clouds())
 		{
 			cmd.print(QObject::tr("\tProcessing cloud %1").arg(!desc.pc->getName().isEmpty() ? desc.pc->getName() : "no name"));
 
-			if (isPercent)
-			{
-				size_t nrOfPoints = desc.pc->size();
-				count = static_cast<unsigned>(ceil(nrOfPoints * percent / 100));
-				cmd.print(QObject::tr("\tOutput points: %1 * %2% = %3").arg(nrOfPoints).arg(percent).arg(count));
+			CCCoreLib::ReferenceCloud* refCloud = nullptr;
+			try {
```

Comment:
can you move the { symbols to the next lines? (as per CC's style).

(by the way, I'm not sure in which state your attempt to activate a global .clangformat file is?)

Reply:
More or less abandoned for now, I should try it again

I think I had some problems between different clang version and the fact that some dependencies are vendorer, but since its been a few months I don't actually remember the specifics